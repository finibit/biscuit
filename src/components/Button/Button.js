import React from 'react'
import PropTypes from 'prop-types'
import styled, { css } from 'styled-components'
import styles from '../../styles'
import types from '../../types'
import { themeGet } from '../../themes'

const typographyStyles = css`
	text-decoration: none;
	font-family: ${themeGet.fontFamily('primary')};
	font-weight: ${themeGet.fontWeight('normal')};
	font-size: ${themeGet.fontSize(1)};
	line-height: ${themeGet.lineHeight('lg')};
`

const normalStyles = css`
	color: ${props => themeGet.colorText(props.color)};
	background-color: ${props => themeGet.color(props.color)};

	&:hover {
		cursor: pointer;
		background-color: ${props => themeGet.colorShade(props.color, .05)};
	}

	&:focus {
		box-shadow: 0px 0px 0px 2px ${props => themeGet.colorShadeOrTint(props.color, .05, .7)};
	}
`

const disabledStyles = css`
	color: ${props => themeGet.colorText(props.color, .5)};
	background-color: ${props => themeGet.colorTint(props.color, .5)};
`

const appearanceStyles = css`
	${props => props.disabled ? disabledStyles : normalStyles};
	border: none;
	border-radius: ${themeGet.borderRadius('md')};
	box-shadow: none;
	text-align: center;
	vertical-align: middle;
	&::-moz-focus-inner { border: 0; }
	&:focus { outline: none; }
	padding: ${props => themeGet.padding(1)} ${props => themeGet.padding(2)};
`

const ButtonStyled = styled.button`
	display: inline-block;
	box-sizing: border-box;
	${typographyStyles}
	${appearanceStyles}

	${styles.compose(
		styles.margin,
	)};
`

const Button = (props) => (
	<ButtonStyled
		{...props}
	/>
)

Button.propTypes = {
	...types.base,
	...types.disabled,
	...types.margin,
	color: PropTypes.string,
}

Button.defaultProps = {
	color: 'primary',
}

export default Button
