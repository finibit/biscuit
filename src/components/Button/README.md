### Usage

#### Type

```jsx
<Button>Default</Button>
<Button type="primary">Primary</Button>
<Button type="info">Info</Button>
<Button type="danger">Danger</Button>
<Button type="success">Success</Button>
<Button type="warning">Warning</Button>
```

### Margin

```jsx
<Button type="primary" margin={{ right: 3 }}>Primary</Button>
<Button type="default">Default</Button>
```

### Padding

```jsx
<Button padding="20px">Default</Button>
```
