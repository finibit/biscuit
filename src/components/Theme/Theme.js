import React from 'react'
import PropTypes from 'prop-types'
import { ThemeProvider, createGlobalStyle } from 'styled-components'
import { mergeTheme, baseTheme, themeGet } from '../../themes'
import types from '../../types'

const GlobalStyle = createGlobalStyle`
	html, body {
		width: 100%;
		height: 100%;
		margin: 0;
		padding: 0;
	}

	body {
		color: ${themeGet.color('black', 'inherit')};
		font-family: ${themeGet.fontFamily('secondary')};
		-webkit-font-smoothing: antialiased;
	}
`

const Theme = (props) => {
	const {
		theme,
		global,
	} = props

	return (
		<ThemeProvider theme={mergeTheme(baseTheme, theme)}>
			<div>
				{ global && <GlobalStyle /> }
				{props.children}
			</div>
		</ThemeProvider>
	)
}

Theme.propTypes = {
	global: PropTypes.bool,
	...types.theme,
}

Theme.defaultProps = {
	global: true,
	theme: baseTheme,
}

export default Theme
