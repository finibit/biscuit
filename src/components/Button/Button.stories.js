import React from 'react'
import { storiesOf } from '@storybook/react'
import Wrapper from '../../Wrapper'
import Button from './Button'

storiesOf('Button', module)
  .addDecorator(Wrapper)
  .add('with color', () => (
    <>
      <Button color='black' mr={1} mb={1}>Black</Button>
      <Button color='white' mr={1} mb={1}>White</Button>
      <Button color='light' mr={1} mb={1}>Light</Button>
      <Button color='primary' mr={1} mb={1}>Primary</Button>
      <Button color='info' mr={1} mb={1}>Info</Button>
      <Button color='danger' mr={1} mb={1}>Danger</Button>
      <Button color='success' mr={1} mb={1}>Success</Button>
      <Button color='warning' mr={1} mb={1}>Warning</Button>
    </>
  ))
  .add('disabled', () => (
    <>
      <Button color='black' mr={1} mb={1} disabled>Black</Button>
      <Button color='white' mr={1} mb={1} disabled>White</Button>
      <Button color='light' mr={1} mb={1} disabled>Light</Button>
      <Button color='primary' mr={1} mb={1} disabled>Primary</Button>
      <Button color='info' mr={1} mb={1} disabled>Info</Button>
      <Button color='danger' mr={1} mb={1} disabled>Danger</Button>
      <Button color='success' mr={1} mb={1} disabled>Success</Button>
      <Button color='warning' mr={1} mb={1} disabled>Warning</Button>
    </>
  ))
  .add('with margin', () => (
    <>
      <Button color='black' m={2}>Black</Button>
      <Button color='white' m={2}>White</Button>
      <Button color='light' m={2}>Light</Button>
      <Button color='primary' m={2}>Primary</Button>
      <Button color='info' m={2}>Info</Button>
      <Button color='danger' m={2}>Danger</Button>
      <Button color='success' m={2}>Success</Button>
      <Button color='warning' m={2}>Warning</Button>
    </>
  ))
