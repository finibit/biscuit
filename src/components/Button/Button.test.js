import React from 'react'
import renderer from 'react-test-renderer'
import 'jest-styled-components'

import Theme from '../Theme'
import Button from './Button'

describe('Button', () => {
	test('with color', () => {
		const component = renderer.create((
			<Theme>
				<Button color="black" />
				<Button color="white" />
				<Button color="light" />
				<Button color="primary" />
				<Button color="info" />
				<Button color="danger" />
				<Button color="success" />
				<Button color="warning" />
			</Theme>
		))
		expect(component.toJSON()).toMatchSnapshot()
	})

	test('disabled', () => {
		const component = renderer.create((
			<Theme>
				<Button color="black" disabled />
				<Button color="white" disabled />
				<Button color="light" disabled />
				<Button color="primary" disabled />
				<Button color="info" disabled />
				<Button color="danger" disabled />
				<Button color="success" disabled />
				<Button color="warning" disabled />
			</Theme>
		))
		expect(component.toJSON()).toMatchSnapshot()
	})

	test('with margin', () => {
		const component = renderer.create((
			<Theme>
				<Button m={1} />
				<Button m={32} />
				<Button m="5%" />
				<Button mt={1} />
				<Button mt={32} />
				<Button mt="5%" />
				<Button ml={1} />
				<Button ml={32} />
				<Button ml="5%" />
				<Button mb={1} />
				<Button mb={32} />
				<Button mb="5%" />
				<Button mr={1} />
				<Button mr={32} />
				<Button mr="5%" />
			</Theme>
		))
		expect(component.toJSON()).toMatchSnapshot()
	})
})
