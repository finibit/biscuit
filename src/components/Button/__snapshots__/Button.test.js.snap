// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button disabled 1`] = `
.c0 {
  display: inline-block;
  box-sizing: border-box;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif;
  font-weight: 400;
  font-size: 1rem;
  line-height: 1.25;
  color: #ffffff;
  background-color: #8f9397;
  border: none;
  border-radius: 4px;
  box-shadow: none;
  text-align: center;
  vertical-align: middle;
  padding: .375em .75em;
}

.c0::-moz-focus-inner {
  border: 0;
}

.c0:focus {
  outline: none;
}

.c1 {
  display: inline-block;
  box-sizing: border-box;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif;
  font-weight: 400;
  font-size: 1rem;
  line-height: 1.25;
  color: #8f9397;
  background-color: #ffffff;
  border: none;
  border-radius: 4px;
  box-shadow: none;
  text-align: center;
  vertical-align: middle;
  padding: .375em .75em;
}

.c1::-moz-focus-inner {
  border: 0;
}

.c1:focus {
  outline: none;
}

.c2 {
  display: inline-block;
  box-sizing: border-box;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif;
  font-weight: 400;
  font-size: 1rem;
  line-height: 1.25;
  color: #8f9397;
  background-color: #f8f9fb;
  border: none;
  border-radius: 4px;
  box-shadow: none;
  text-align: center;
  vertical-align: middle;
  padding: .375em .75em;
}

.c2::-moz-focus-inner {
  border: 0;
}

.c2:focus {
  outline: none;
}

.c3 {
  display: inline-block;
  box-sizing: border-box;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif;
  font-weight: 400;
  font-size: 1rem;
  line-height: 1.25;
  color: #8f9397;
  background-color: #ffeeac;
  border: none;
  border-radius: 4px;
  box-shadow: none;
  text-align: center;
  vertical-align: middle;
  padding: .375em .75em;
}

.c3::-moz-focus-inner {
  border: 0;
}

.c3:focus {
  outline: none;
}

.c4 {
  display: inline-block;
  box-sizing: border-box;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif;
  font-weight: 400;
  font-size: 1rem;
  line-height: 1.25;
  color: #ffffff;
  background-color: #87defc;
  border: none;
  border-radius: 4px;
  box-shadow: none;
  text-align: center;
  vertical-align: middle;
  padding: .375em .75em;
}

.c4::-moz-focus-inner {
  border: 0;
}

.c4:focus {
  outline: none;
}

.c5 {
  display: inline-block;
  box-sizing: border-box;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif;
  font-weight: 400;
  font-size: 1rem;
  line-height: 1.25;
  color: #ffffff;
  background-color: #ff9f9a;
  border: none;
  border-radius: 4px;
  box-shadow: none;
  text-align: center;
  vertical-align: middle;
  padding: .375em .75em;
}

.c5::-moz-focus-inner {
  border: 0;
}

.c5:focus {
  outline: none;
}

.c6 {
  display: inline-block;
  box-sizing: border-box;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif;
  font-weight: 400;
  font-size: 1rem;
  line-height: 1.25;
  color: #ffffff;
  background-color: #82e2b5;
  border: none;
  border-radius: 4px;
  box-shadow: none;
  text-align: center;
  vertical-align: middle;
  padding: .375em .75em;
}

.c6::-moz-focus-inner {
  border: 0;
}

.c6:focus {
  outline: none;
}

.c7 {
  display: inline-block;
  box-sizing: border-box;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif;
  font-weight: 400;
  font-size: 1rem;
  line-height: 1.25;
  color: #ffffff;
  background-color: #ffd480;
  border: none;
  border-radius: 4px;
  box-shadow: none;
  text-align: center;
  vertical-align: middle;
  padding: .375em .75em;
}

.c7::-moz-focus-inner {
  border: 0;
}

.c7:focus {
  outline: none;
}

<div>
  <button
    className="c0"
    color="black"
    disabled={true}
  />
  <button
    className="c1"
    color="white"
    disabled={true}
  />
  <button
    className="c2"
    color="light"
    disabled={true}
  />
  <button
    className="c3"
    color="primary"
    disabled={true}
  />
  <button
    className="c4"
    color="info"
    disabled={true}
  />
  <button
    className="c5"
    color="danger"
    disabled={true}
  />
  <button
    className="c6"
    color="success"
    disabled={true}
  />
  <button
    className="c7"
    color="warning"
    disabled={true}
  />
</div>
`;

exports[`Button with color 1`] = `
.c0 {
  display: inline-block;
  box-sizing: border-box;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif;
  font-weight: 400;
  font-size: 1rem;
  line-height: 1.25;
  color: #ffffff;
  background-color: #1e272e;
  border: none;
  border-radius: 4px;
  box-shadow: none;
  text-align: center;
  vertical-align: middle;
  padding: .375em .75em;
}

.c0:hover {
  cursor: pointer;
  background-color: #1d252c;
}

.c0:focus {
  box-shadow: 0px 0px 0px 2px #bcbec0;
}

.c0::-moz-focus-inner {
  border: 0;
}

.c0:focus {
  outline: none;
}

.c1 {
  display: inline-block;
  box-sizing: border-box;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif;
  font-weight: 400;
  font-size: 1rem;
  line-height: 1.25;
  color: #1e272e;
  background-color: #ffffff;
  border: none;
  border-radius: 4px;
  box-shadow: none;
  text-align: center;
  vertical-align: middle;
  padding: .375em .75em;
}

.c1:hover {
  cursor: pointer;
  background-color: #f2f2f2;
}

.c1:focus {
  box-shadow: 0px 0px 0px 2px #f2f2f2;
}

.c1::-moz-focus-inner {
  border: 0;
}

.c1:focus {
  outline: none;
}

.c2 {
  display: inline-block;
  box-sizing: border-box;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif;
  font-weight: 400;
  font-size: 1rem;
  line-height: 1.25;
  color: #1e272e;
  background-color: #f1f2f6;
  border: none;
  border-radius: 4px;
  box-shadow: none;
  text-align: center;
  vertical-align: middle;
  padding: .375em .75em;
}

.c2:hover {
  cursor: pointer;
  background-color: #e5e6ea;
}

.c2:focus {
  box-shadow: 0px 0px 0px 2px #e5e6ea;
}

.c2::-moz-focus-inner {
  border: 0;
}

.c2:focus {
  outline: none;
}

.c3 {
  display: inline-block;
  box-sizing: border-box;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif;
  font-weight: 400;
  font-size: 1rem;
  line-height: 1.25;
  color: #1e272e;
  background-color: #ffdd59;
  border: none;
  border-radius: 4px;
  box-shadow: none;
  text-align: center;
  vertical-align: middle;
  padding: .375em .75em;
}

.c3:hover {
  cursor: pointer;
  background-color: #f2d255;
}

.c3:focus {
  box-shadow: 0px 0px 0px 2px #f2d255;
}

.c3::-moz-focus-inner {
  border: 0;
}

.c3:focus {
  outline: none;
}

.c4 {
  display: inline-block;
  box-sizing: border-box;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif;
  font-weight: 400;
  font-size: 1rem;
  line-height: 1.25;
  color: #ffffff;
  background-color: #0fbcf9;
  border: none;
  border-radius: 4px;
  box-shadow: none;
  text-align: center;
  vertical-align: middle;
  padding: .375em .75em;
}

.c4:hover {
  cursor: pointer;
  background-color: #0eb3ed;
}

.c4:focus {
  box-shadow: 0px 0px 0px 2px #b7ebfd;
}

.c4::-moz-focus-inner {
  border: 0;
}

.c4:focus {
  outline: none;
}

.c5 {
  display: inline-block;
  box-sizing: border-box;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif;
  font-weight: 400;
  font-size: 1rem;
  line-height: 1.25;
  color: #ffffff;
  background-color: #ff3f34;
  border: none;
  border-radius: 4px;
  box-shadow: none;
  text-align: center;
  vertical-align: middle;
  padding: .375em .75em;
}

.c5:hover {
  cursor: pointer;
  background-color: #f23c31;
}

.c5:focus {
  box-shadow: 0px 0px 0px 2px #ffc5c2;
}

.c5::-moz-focus-inner {
  border: 0;
}

.c5:focus {
  outline: none;
}

.c6 {
  display: inline-block;
  box-sizing: border-box;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif;
  font-weight: 400;
  font-size: 1rem;
  line-height: 1.25;
  color: #ffffff;
  background-color: #05c46b;
  border: none;
  border-radius: 4px;
  box-shadow: none;
  text-align: center;
  vertical-align: middle;
  padding: .375em .75em;
}

.c6:hover {
  cursor: pointer;
  background-color: #05ba66;
}

.c6:focus {
  box-shadow: 0px 0px 0px 2px #b4edd3;
}

.c6::-moz-focus-inner {
  border: 0;
}

.c6:focus {
  outline: none;
}

.c7 {
  display: inline-block;
  box-sizing: border-box;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif;
  font-weight: 400;
  font-size: 1rem;
  line-height: 1.25;
  color: #ffffff;
  background-color: #ffa801;
  border: none;
  border-radius: 4px;
  box-shadow: none;
  text-align: center;
  vertical-align: middle;
  padding: .375em .75em;
}

.c7:hover {
  cursor: pointer;
  background-color: #f2a001;
}

.c7:focus {
  box-shadow: 0px 0px 0px 2px #ffe5b3;
}

.c7::-moz-focus-inner {
  border: 0;
}

.c7:focus {
  outline: none;
}

<div>
  <button
    className="c0"
    color="black"
  />
  <button
    className="c1"
    color="white"
  />
  <button
    className="c2"
    color="light"
  />
  <button
    className="c3"
    color="primary"
  />
  <button
    className="c4"
    color="info"
  />
  <button
    className="c5"
    color="danger"
  />
  <button
    className="c6"
    color="success"
  />
  <button
    className="c7"
    color="warning"
  />
</div>
`;

exports[`Button with margin 1`] = `
.c0 {
  display: inline-block;
  box-sizing: border-box;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif;
  font-weight: 400;
  font-size: 1rem;
  line-height: 1.25;
  color: #1e272e;
  background-color: #ffdd59;
  border: none;
  border-radius: 4px;
  box-shadow: none;
  text-align: center;
  vertical-align: middle;
  padding: .375em .75em;
  margin: 0.375rem;
}

.c0:hover {
  cursor: pointer;
  background-color: #f2d255;
}

.c0:focus {
  box-shadow: 0px 0px 0px 2px #f2d255;
}

.c0::-moz-focus-inner {
  border: 0;
}

.c0:focus {
  outline: none;
}

.c1 {
  display: inline-block;
  box-sizing: border-box;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif;
  font-weight: 400;
  font-size: 1rem;
  line-height: 1.25;
  color: #1e272e;
  background-color: #ffdd59;
  border: none;
  border-radius: 4px;
  box-shadow: none;
  text-align: center;
  vertical-align: middle;
  padding: .375em .75em;
  margin: 32px;
}

.c1:hover {
  cursor: pointer;
  background-color: #f2d255;
}

.c1:focus {
  box-shadow: 0px 0px 0px 2px #f2d255;
}

.c1::-moz-focus-inner {
  border: 0;
}

.c1:focus {
  outline: none;
}

.c2 {
  display: inline-block;
  box-sizing: border-box;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif;
  font-weight: 400;
  font-size: 1rem;
  line-height: 1.25;
  color: #1e272e;
  background-color: #ffdd59;
  border: none;
  border-radius: 4px;
  box-shadow: none;
  text-align: center;
  vertical-align: middle;
  padding: .375em .75em;
  margin: 5%;
}

.c2:hover {
  cursor: pointer;
  background-color: #f2d255;
}

.c2:focus {
  box-shadow: 0px 0px 0px 2px #f2d255;
}

.c2::-moz-focus-inner {
  border: 0;
}

.c2:focus {
  outline: none;
}

.c3 {
  display: inline-block;
  box-sizing: border-box;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif;
  font-weight: 400;
  font-size: 1rem;
  line-height: 1.25;
  color: #1e272e;
  background-color: #ffdd59;
  border: none;
  border-radius: 4px;
  box-shadow: none;
  text-align: center;
  vertical-align: middle;
  padding: .375em .75em;
  margin-top: 0.375rem;
}

.c3:hover {
  cursor: pointer;
  background-color: #f2d255;
}

.c3:focus {
  box-shadow: 0px 0px 0px 2px #f2d255;
}

.c3::-moz-focus-inner {
  border: 0;
}

.c3:focus {
  outline: none;
}

.c4 {
  display: inline-block;
  box-sizing: border-box;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif;
  font-weight: 400;
  font-size: 1rem;
  line-height: 1.25;
  color: #1e272e;
  background-color: #ffdd59;
  border: none;
  border-radius: 4px;
  box-shadow: none;
  text-align: center;
  vertical-align: middle;
  padding: .375em .75em;
  margin-top: 32px;
}

.c4:hover {
  cursor: pointer;
  background-color: #f2d255;
}

.c4:focus {
  box-shadow: 0px 0px 0px 2px #f2d255;
}

.c4::-moz-focus-inner {
  border: 0;
}

.c4:focus {
  outline: none;
}

.c5 {
  display: inline-block;
  box-sizing: border-box;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif;
  font-weight: 400;
  font-size: 1rem;
  line-height: 1.25;
  color: #1e272e;
  background-color: #ffdd59;
  border: none;
  border-radius: 4px;
  box-shadow: none;
  text-align: center;
  vertical-align: middle;
  padding: .375em .75em;
  margin-top: 5%;
}

.c5:hover {
  cursor: pointer;
  background-color: #f2d255;
}

.c5:focus {
  box-shadow: 0px 0px 0px 2px #f2d255;
}

.c5::-moz-focus-inner {
  border: 0;
}

.c5:focus {
  outline: none;
}

.c6 {
  display: inline-block;
  box-sizing: border-box;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif;
  font-weight: 400;
  font-size: 1rem;
  line-height: 1.25;
  color: #1e272e;
  background-color: #ffdd59;
  border: none;
  border-radius: 4px;
  box-shadow: none;
  text-align: center;
  vertical-align: middle;
  padding: .375em .75em;
  margin-left: 0.375rem;
}

.c6:hover {
  cursor: pointer;
  background-color: #f2d255;
}

.c6:focus {
  box-shadow: 0px 0px 0px 2px #f2d255;
}

.c6::-moz-focus-inner {
  border: 0;
}

.c6:focus {
  outline: none;
}

.c7 {
  display: inline-block;
  box-sizing: border-box;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif;
  font-weight: 400;
  font-size: 1rem;
  line-height: 1.25;
  color: #1e272e;
  background-color: #ffdd59;
  border: none;
  border-radius: 4px;
  box-shadow: none;
  text-align: center;
  vertical-align: middle;
  padding: .375em .75em;
  margin-left: 32px;
}

.c7:hover {
  cursor: pointer;
  background-color: #f2d255;
}

.c7:focus {
  box-shadow: 0px 0px 0px 2px #f2d255;
}

.c7::-moz-focus-inner {
  border: 0;
}

.c7:focus {
  outline: none;
}

.c8 {
  display: inline-block;
  box-sizing: border-box;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif;
  font-weight: 400;
  font-size: 1rem;
  line-height: 1.25;
  color: #1e272e;
  background-color: #ffdd59;
  border: none;
  border-radius: 4px;
  box-shadow: none;
  text-align: center;
  vertical-align: middle;
  padding: .375em .75em;
  margin-left: 5%;
}

.c8:hover {
  cursor: pointer;
  background-color: #f2d255;
}

.c8:focus {
  box-shadow: 0px 0px 0px 2px #f2d255;
}

.c8::-moz-focus-inner {
  border: 0;
}

.c8:focus {
  outline: none;
}

.c9 {
  display: inline-block;
  box-sizing: border-box;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif;
  font-weight: 400;
  font-size: 1rem;
  line-height: 1.25;
  color: #1e272e;
  background-color: #ffdd59;
  border: none;
  border-radius: 4px;
  box-shadow: none;
  text-align: center;
  vertical-align: middle;
  padding: .375em .75em;
  margin-bottom: 0.375rem;
}

.c9:hover {
  cursor: pointer;
  background-color: #f2d255;
}

.c9:focus {
  box-shadow: 0px 0px 0px 2px #f2d255;
}

.c9::-moz-focus-inner {
  border: 0;
}

.c9:focus {
  outline: none;
}

.c10 {
  display: inline-block;
  box-sizing: border-box;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif;
  font-weight: 400;
  font-size: 1rem;
  line-height: 1.25;
  color: #1e272e;
  background-color: #ffdd59;
  border: none;
  border-radius: 4px;
  box-shadow: none;
  text-align: center;
  vertical-align: middle;
  padding: .375em .75em;
  margin-bottom: 32px;
}

.c10:hover {
  cursor: pointer;
  background-color: #f2d255;
}

.c10:focus {
  box-shadow: 0px 0px 0px 2px #f2d255;
}

.c10::-moz-focus-inner {
  border: 0;
}

.c10:focus {
  outline: none;
}

.c11 {
  display: inline-block;
  box-sizing: border-box;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif;
  font-weight: 400;
  font-size: 1rem;
  line-height: 1.25;
  color: #1e272e;
  background-color: #ffdd59;
  border: none;
  border-radius: 4px;
  box-shadow: none;
  text-align: center;
  vertical-align: middle;
  padding: .375em .75em;
  margin-bottom: 5%;
}

.c11:hover {
  cursor: pointer;
  background-color: #f2d255;
}

.c11:focus {
  box-shadow: 0px 0px 0px 2px #f2d255;
}

.c11::-moz-focus-inner {
  border: 0;
}

.c11:focus {
  outline: none;
}

.c12 {
  display: inline-block;
  box-sizing: border-box;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif;
  font-weight: 400;
  font-size: 1rem;
  line-height: 1.25;
  color: #1e272e;
  background-color: #ffdd59;
  border: none;
  border-radius: 4px;
  box-shadow: none;
  text-align: center;
  vertical-align: middle;
  padding: .375em .75em;
  margin-right: 0.375rem;
}

.c12:hover {
  cursor: pointer;
  background-color: #f2d255;
}

.c12:focus {
  box-shadow: 0px 0px 0px 2px #f2d255;
}

.c12::-moz-focus-inner {
  border: 0;
}

.c12:focus {
  outline: none;
}

.c13 {
  display: inline-block;
  box-sizing: border-box;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif;
  font-weight: 400;
  font-size: 1rem;
  line-height: 1.25;
  color: #1e272e;
  background-color: #ffdd59;
  border: none;
  border-radius: 4px;
  box-shadow: none;
  text-align: center;
  vertical-align: middle;
  padding: .375em .75em;
  margin-right: 32px;
}

.c13:hover {
  cursor: pointer;
  background-color: #f2d255;
}

.c13:focus {
  box-shadow: 0px 0px 0px 2px #f2d255;
}

.c13::-moz-focus-inner {
  border: 0;
}

.c13:focus {
  outline: none;
}

.c14 {
  display: inline-block;
  box-sizing: border-box;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif;
  font-weight: 400;
  font-size: 1rem;
  line-height: 1.25;
  color: #1e272e;
  background-color: #ffdd59;
  border: none;
  border-radius: 4px;
  box-shadow: none;
  text-align: center;
  vertical-align: middle;
  padding: .375em .75em;
  margin-right: 5%;
}

.c14:hover {
  cursor: pointer;
  background-color: #f2d255;
}

.c14:focus {
  box-shadow: 0px 0px 0px 2px #f2d255;
}

.c14::-moz-focus-inner {
  border: 0;
}

.c14:focus {
  outline: none;
}

<div>
  <button
    className="c0"
    color="primary"
  />
  <button
    className="c1"
    color="primary"
  />
  <button
    className="c2"
    color="primary"
  />
  <button
    className="c3"
    color="primary"
  />
  <button
    className="c4"
    color="primary"
  />
  <button
    className="c5"
    color="primary"
  />
  <button
    className="c6"
    color="primary"
  />
  <button
    className="c7"
    color="primary"
  />
  <button
    className="c8"
    color="primary"
  />
  <button
    className="c9"
    color="primary"
  />
  <button
    className="c10"
    color="primary"
  />
  <button
    className="c11"
    color="primary"
  />
  <button
    className="c12"
    color="primary"
  />
  <button
    className="c13"
    color="primary"
  />
  <button
    className="c14"
    color="primary"
  />
</div>
`;
